### PR TITLE
add pylint configuration

### PR DESCRIPTION
**Pull request by Artturin**
but not tests yet

### CODE REVIEW

**Comment by Artturin** [comment]
Will fix lints in a separate PR

**Comment by figsoda** [comment]
Ruff seems to support some pylint lints under `PL`, since we are already using ruff, can we use ruff instead? Adding `"PL"` to `tools.ruff.select` might be sufficient

File: nixpkgs_review/nix.py [hunk 1]
```diff
@@ -297,7 +297,7 @@ def nix_build(
 def write_shell_expression(
     filename: Path, attrs: list[str], system: str, nixpkgs_config: Path
 ) -> None:
-    with open(filename, "w+") as f:
+    with open(filename, "w+", encoding="utf-8") as f:
```
Comment (Mic92) [inline nixpkgs_review/nix.py:300]: What happens if you don't specify encoding?

**Comment by Mic92** [comment]
Here is a feature comparison: https://github.com/astral-sh/ruff/blob/d31a700f773b97b265109b63bbeb0ef994f63a3b/README.md#how-does-ruff-compare-to-pylint

**Comment by Mic92** [comment]
I guess it doesn't hurt.. checking should be still fast enough

**Comment by Mic92** [comment]
@mergify queue

File: nixpkgs_review/nix.py (hunk 1 above)
Comment (Artturin) [reply nixpkgs_review/nix.py:300]: https://pylint.pycqa.org/en/latest/user_guide/messages/warning/unspecified-encoding.html

> It is better to specify an encoding when opening documents. Using the system default implicitly can create problems on other operating systems. See https://peps.python.org/pep-0597/

**Pull request merged**